recipe-client-addon/lib/EventEmitter.jsm

```diff
@@ -20,15 +19,13 @@ this.EventEmitter = function(driver) {
       Promise.resolve()
         .then(() => {
           if (!(eventName in listeners)) {
-            driver.log(`EventEmitter: Event fired with no listeners: ${eventName}`);
+            log.info(`EventEmitter: Event fired with no listeners: ${eventName}`);
             return;
           }
-          // freeze event to prevent handlers from modifying it
-          const frozenEvent = Object.freeze(event);
```

Comment:
I blame @jimblandy for telling me this was a thing, and myself for not properly testing it.

The problem with freezing as it was before is that the cloned object passed in to `emit` is a proxy object (because it has been cloned into another compartment, so we get an xrayed wrapper? I think) and throws when you try to freeze it. 

So we need to freeze inside of the sandbox, but IIRC there were concerns with the EventEmitter class having a reference to the sandbox and also being cloned into it due to memory leaks and/or potential security issues. So I was hoping to avoid the EventEmitter class itself having a reference to the sandbox.

I'll see if I can find a way around this, and will report back.